### PR TITLE
daml-sdk-head: Delete the SDK temporary directory after installation.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
I'm aggregating a lot of SDKs.

```
$ du -h -d1 $TMPDIR | sort -rh | head -5
1.9G    /var/folders/5x/36ws6_n96qz_b4tbpvfbcth80000gn/T/
501M    /var/folders/5x/36ws6_n96qz_b4tbpvfbcth80000gn/T/tmp.ahBTENEeZw
500M    /var/folders/5x/36ws6_n96qz_b4tbpvfbcth80000gn/T/tmp.S4sB7DIjkA
500M    /var/folders/5x/36ws6_n96qz_b4tbpvfbcth80000gn/T/tmp.Q3VeVV8DIU
405M    /var/folders/5x/36ws6_n96qz_b4tbpvfbcth80000gn/T/tmp.b5ZWnPhLkH
```

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: dev-env/bin/daml-sdk-head [hunk 1]
```diff
@@ -97,28 +97,37 @@ echo ""
...
 }
 trap cleanup EXIT
 
 # Building here separately so the user can see the build process which could take a while
 bazel build "${BAZEL_MODE_FLAGS[@]}" //release:sdk-release-tarball.tar.gz
 
 readonly TARBALL="$(bazel info bazel-bin "${BAZEL_MODE_FLAGS[@]}")/release/sdk-release-tarball.tar.gz"
-readonly TMPDIR="$(mktemp -d)"
-mkdir -p "${TMPDIR}/sdk-head"
+readonly SDK_TEMP_DIR="$(mktemp -d)"
```
Comment (cocreature) [inline dev-env/bin/daml-sdk-head:111]: the variable names here seem a bit confusing but I don’t have a better suggestion either.

File: dev-env/bin/daml-sdk-head [hunk 2]
```diff
@@ -97,28 +97,37 @@ echo ""
...
-  "${DAML_HOME}/bin/daml" install "${TMPDIR}/sdk-head" --force
+  "${DAML_HOME}/bin/daml" install "$SDK_DIR" --force
 else
   # No daml installation detected, so install the tarball normally but disable auto-install.
-  "${TMPDIR}/sdk-head/install.sh" --force
+  "${SDK_DIR}/install.sh" --force
   echo "auto-install: false" > "${DAML_HOME}/daml-config.yaml"
 fi
 
+if [[ -d "$SDK_TEMP_DIR" ]]; then
```
Comment (cocreature) [inline dev-env/bin/daml-sdk-head:127]: Any reason not to `trap` this? Just because we already have other traps so it’s a bit annoying?

Comment (SamirTalwar-DA) [reply dev-env/bin/daml-sdk-head:127]: It's trapped above, but we remove that trap a little later, and I don't think we can have multiple traps (and only remove one at a time). If we can, happy to do that.

File: dev-env/bin/daml-sdk-head (hunk 1 above)
Comment (SamirTalwar-DA) [reply dev-env/bin/daml-sdk-head:111]: Agreed. If you can think of anything, let me know.

**Pull request merged**